bootstrap4/.js/src/main/scala/io/udash/bootstrap/datepicker/UdashDatePicker.scala

```diff
@@ -537,37 +559,53 @@ object UdashDatePicker {
     def valueOf(): Double = js.native
   }
 
-  import org.scalajs.dom.raw.{MutationObserver, MutationObserverInit, MutationRecord}
+  import org.scalajs.dom.raw.{MutationObserver, MutationObserverInit, MutationRecord, NodeList}
+
   import scala.collection.mutable.{Map => MMap}
 
   private val datePickerSetupCallbacks = MMap.empty[ComponentId, () => Unit]
+  private val datePickerDetachCallbacks = MMap.empty[ComponentId, () => Unit]
 
   // When a date picker gets appended to a DOM, its options and listeners should be initialized once again. Thus, all
   // nodes with mutated children are examined whether there is a date picker defined within a corresponding DOM
   // subtree. It is also substantially possible for any date picker to be reported by more than one `MutationRecord`
   // since the `subtree` switch of `MutationObserverInit` is on, thus only the first update is considered for each
-  // picker (note `exists` call) to avoid double initialization. This mutation observer is turned off when there are no
-  // components registered to be watched.
-  private val datePickerMutationObserver =
+  // picker (note `exists` call) to avoid double initialization. Similarly, Tempus Dominus custom jQuery event listeners
+  // get deregistered each time a date picker is detached from the DOM.
+  //
+  // This mutation observer is turned on just before the first callback registration and off when there are no components
+  // registered to be watched.
+  private val datePickerMutationObserver = {
     new MutationObserver((records: js.Array[MutationRecord], _: MutationObserver) => {
-      val addedNodes = records.flatMap(record => for {i <- 0 until record.addedNodes.length} yield record.addedNodes(i))
-      datePickerSetupCallbacks.foreach { case (pickerId, callback) =>
-        if (addedNodes.exists {
-          case element: Element => element.id == pickerId.id || element.querySelector(s"#$pickerId") != null
-          case _ => false
-        }) callback()
+      def mutationHandler(nodesExtractor: MutationRecord => NodeList, callbacks: MMap[ComponentId, () => Unit]): Unit = {
+        val nodes = records.flatMap(record =>
```

Comment:
Alternative approach:
- register on rendered nodes instead of IDs to avoid query selector
- check if registered nodes are equal or children of removed / added nodes via `Node#contains`